Add tagger section to README per 1272

x1101 opened the pull request:
Adding how-to on release tagger per #1272

Happy for feedback on language usage, tone, etc as well as technical details.

x1101 committed:
First pass at adding tagging doc

x1101 committed:
Further Doc Cleanup

x1101 committed:
Add Details

oraNod commented:
Hey @x1101 welcome! Thanks for the PR. :rocket:

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
 podman run --rm --tty --volume "$(pwd):/mnt:z" --workdir /mnt docker.io/library/python:3.10 bash -c 'pip install nox ; nox -s pip-compile'
 ```
+
+## Creating release tags
````
```suggestion
## Creating release tags

```

x1101 replied:
Is this just "new line following a heading" for Markdown? I guess I need to get my MD linting setup to catch this. Sorry.

oraNod replied:
Yeah, just a new line after the heading. No worries.

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
 ```
+
+## Creating release tags
+Creating tags that correspond with releases and release candidates in [`ansible/ansible`](https://github.com/ansible/ansible)
````
```suggestion
When a tag is created in the [`ansible/ansible`](https://github.com/ansible/ansible) repository for a release or release candidate, a corresponding tag should be created in this `ansible-documentation` repository.
```

oraNod replied:
It's probably a bit wordier but I think we need to explain a bit more clearly "when" to create tags.

x1101 replied:
Agreed. This absolutely makes it more clear.

oraNod commented on README.md:
```diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+## Creating release tags
+Creating tags that correspond with releases and release candidates in [`ansible/ansible`](https://github.com/ansible/ansible)
+
+First, install the additional tagging dependencies
```
```suggestion
First, install the additional tagging dependencies as follows:
```

oraNod replied:
There needs to be some kind of ending punctuation for the sentence. Because this is a "lead in" or introductory sentence, I tend to use "as follows:" but you could reword it as you see fit.

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+First, install the additional tagging dependencies
+
+```bash
+pip install gitpython typer
````
```suggestion
python -m pip install -r hacking/tagger/requirements.txt
```

oraNod replied:
There's a requirements file in the `tagger` folder that you can use. It might also be an idea to suggest creating a venv and installing into that to avoid polluting the user space.

x1101 replied:
Yeah, this was the bit I'd had questions about if I'd done right. I'd initially followed [the manual setup guide](https://docs.ansible.com/ansible/latest/community/documentation_contributions.html#setting-up-your-environment-to-build-documentation-locally), and then looked at the `nox` setup as well, but neither got the needed requirements. 

I'd written this part *assuming* that the reader had done one of those at this point (which I shouldn't do). 

Is there something cleaner than this? If not, I'll update it to at least suggest creating a venv.

oraNod replied:
Not quite sure what you mean by cleaner. I suppose you don't really need the `python -m` portion and can trim it down to just `pip install`.

The venv bit can just be a comment in the codeblock too so it's discrete. I'd imagine anyone doing this would have a venv already so just a hint is all that's needed really.

x1101 replied:
Do you think that + the updated info above makes this clear enough, and has sufficient reminder about the venv?

oraNod replied:
Yes, I think that's fine now. Thank you.

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+pip install gitpython typer
+```
+
+Next, you'll need to ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) checked out in sibling directories.
````
```suggestion
Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out at the same directory level.
```

oraNod commented on README.md:
```diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Next, you'll need to ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) checked out in sibling directories.
+
+Lastly, run the various tasks with the tagging tool.
```
```suggestion
Lastly, run the appropriate commands with the tagger script.
```

oraNod commented on README.md:
```diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Lastly, run the various tasks with the tagging tool.
+
+Show tags from ansible-core that are not present here:
```
```suggestion
Search for new tags in the `ansible-core` repository:

```

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+./backing/tagger/tag.py tag
+```
+
+See `--help` for extended options
````
```suggestion
See `--help` for extended options.
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+./hacking/tagger/tag.py new-tags
+```
+
+Determine missing `anisble-core` releases from `--branch`. Create corresponding tags for releases in `ansible-documentation`, and push them:
````
TBH I'm not sure what you mean with the first sentence.

x1101 replied:
Yeah, I don't know how to clearly express this. I have the code bases checked out at ~/projects/ansible/ansible and ~/projects/ansible/ansible-documentation (so, they're sibling dirs, at the same level). But `~/projects` is just my convention, and I added `ansible/` in there to keep all related projects together. 

It *seems* like the initial request for this implied that both code bases needed to be checked out like that, but the tool seems to find them. I'll play with it a bit and see how forgiving it is and then work on cleaning this up.

x1101 replied:
After some testing on the taggers auto detection of repository locations, I've updated this section and added a bit more to the "run" bit showing how they'd be used.

oraNod replied:
I was referring to this sentence: "Determine missing `anisble-core` releases from `--branch`."

Could you please explain this a bit more? I don't think the tagger has a `--branch` parameter so I'm not sure what that refers to.

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+./hacking/tagger/tag.py new-tags
+```
+
+Determine missing `anisble-core` releases from `--branch`. Create corresponding tags for releases in `ansible-documentation`, and push them:
````
```suggestion
Determine missing `anisble-core` releases from `--branch`. Create corresponding tags and push them to the `ansible-documentation` repository:
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Determine missing `anisble-core` releases from `--branch`. Create corresponding tags for releases in `ansible-documentation`, and push them:
+```bash
+./backing/tagger/tag.py tag
````
```suggestion
./hacking/tagger/tag.py tag
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,28 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+Determine missing `anisble-core` releases from `--branch`. Create corresponding tags for releases in `ansible-documentation`, and push them:
+```bash
+./backing/tagger/tag.py tag
+```
````
You might also consider adding an example with the `--remote` option because not everyone (including me) sets the upstream as origin. Maybe combine into a single codeblock like this:

```bash
# The tagger script assumes origin as the upstream remote.
./hacking/tagger/tag.py tag

# If you use something else as the upstream remote, specify the name.
./hacking/tagger/tag.py tag --remote <name> 
```

x1101 committed:
Update README.md

Markdown standards fixup

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

wording fixup

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Clearer description for why/when to tag

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

language clarity fixup

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

silly typo fixup

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

clarity fixup

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
For clarity on venv, and reqs

x1101 committed:
clarity on remotes for tagger

x1101 committed:
Markdown quality fixup

oraNod commented on README.md:
````diff
@@ -136,3 +136,37 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+```bash
+python3 -m venv ./venv
+source ./venv/bin/activated 
````
```suggestion
source ./venv/bin/activate
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,37 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+```bash
+python3 -m venv ./venv
+source ./venv/bin/activated 
+pip -r hacking/tagger/requirements.txt
````
```suggestion
pip install -r hacking/tagger/requirements.txt
```

x1101 replied:
Found that as I was walking through it to test the repo detection. Already fixed

x1101 committed:
clarify checkout instructions for tagging

x1101 committed:
Merge branch 'ansible:devel' into devel

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 commented:
At this point I think I've addressed all the comments. I'd appreciate another look at this and any more feedback. 

Big thanks to @oraNod for collaboration and helping me come up to speed.

oraNod commented on README.md:
````diff
@@ -136,3 +136,56 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+```bash
+python3 -m venv ./venv
+source ./venv/bin/activate
+pip insatll -r hacking/tagger/requirements.txt
````
```suggestion
pip install -r hacking/tagger/requirements.txt
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,56 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+pip insatll -r hacking/tagger/requirements.txt
+```
+
+Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out.  The tool assumes that both checkouts are done from the same directory, or you can override this with the `--docs` and `--remote` options. For example:
````
I think the `--remote` option is for the name of the upstream remote, rather than anything to do with the location of the checkout on the filesystem.

For example, here's mine:

```
$ git remote -v
upstream	git@github.com:ansible/ansible-documentation.git (fetch)
upstream	git@github.com:ansible/ansible-documentation.git (push)
```

Whereas the tagger defaults to "origin" as the remote name for upstream.

x1101 replied:
Yeah, that was a typo. Fixed.

oraNod commented on README.md:
```diff
@@ -136,3 +136,56 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+# confirm
+ls -1
+ansible
+ansible-documentation
```
This seems a little too much detail. I think it's better to be concise and just make it clear that both repos need to be checked out at the same level of the filesystem.

x1101 replied:
I pulled it back out. Thanks for the feedback

oraNod commented on README.md:
```diff
@@ -136,3 +136,56 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+./hacking/tagger/tag.py --remote <name> tag 
+
+# If your core repo is in a non-default location
+./hacking/tagger/tag.py --core <location> tag
```
```suggestion
./hacking/tagger/tag.py --core <path> tag
```

x1101 replied:
Fixed

oraNod reviewed (commented):
Thanks for the changes @x1101 I've got a few more comments for you.

x1101 committed:
Update README.md

typo

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Fixups

oraNod commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+./hacking/tagger/tag.py new-tags
+```
+
+Determine missing anisble-core releases. Create corresponding tags for releases in ansible-documentation, and push them:
````
```suggestion
If the preceding command shows that there are new `ansible-core` tags, create corresponding ones for the `ansible-documentation` repository as follows:
```

oraNod commented on README.md:
```diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+# If you use a different upstream remote, specify the name like this
+./hacking/tagger/tag.py --remote <name> tag 
+
+# If your core repo is in a non-default location
```
```suggestion
# If your core repo is not in the same filesystem location, specify the path.
```

oraNod commented on README.md:
```diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+# The tagger scripts assumes "origin" as the upstream remote
+./hacking/tagger/tag.py tag
+
+# If you use a different upstream remote, specify the name like this
```
```suggestion
# If you use a different upstream remote, specify the name.
```

oraNod commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+Determine missing anisble-core releases. Create corresponding tags for releases in ansible-documentation, and push them:
+
+```bash
+# The tagger scripts assumes "origin" as the upstream remote
````
```suggestion
# The tagger scripts assumes "origin" as the upstream remote.
```

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Merge branch 'ansible:devel' into devel

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+python3 -m venv ./venv
+source ./venv/bin/activate
+pip install -r hacking/tagger/requirements.txt
+```
````
We should probably add a nox session to do this instead, but I don't consider that a blocker here.

x1101 replied:
Why don't we add that to the backlog once this PR is complete. I'd be happy to try wrapping my head around that.

samccann replied:
@x1101 that would be great, thanks! We'd welcome a separate PR for that.

oraNod commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+pip install -r hacking/tagger/requirements.txt
+```
+
+Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out.  The tool assumes that both checkouts are done from the same directory, or you can override this with the `--docs` and `--core` options.
````
```suggestion
Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out. The tool assumes that both checkouts are done from the same directory. You can set different paths to your checkouts with the `--docs` and `--core` options.
```

oraNod commented on README.md:
```diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Lastly, run the appropriate command(s) with the tagger script.
+
+Search for tags in the ansible-core repository that are missing from ansible-documenmtation
```
```suggestion
Search for tags in the `ansible-core` repository that are missing from the `ansible-documentation` repository.
```

oraNod reviewed (commented):
Couple final nits. Thanks again for all the work on this one @x1101

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Don Naro <dnaro@redhat.com>

oraNod reviewed (approved):
LGTM. Thanks for the docs contribution @x1101

x1101 committed:
Merge branch 'ansible:devel' into devel

x1101 committed:
Merge branch 'ansible:devel' into devel

samccann commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+```
+
+Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out. The tool assumes that both checkouts are done from the same directory. You can set different paths to your checkouts with the `--docs` and `--core` options.
+
````
As someone who's never done this, I'm trying to see if I can follow these instructions.  I'm unsure what this section means by 'checkout done from same directory'.

So for example, locally, I have `ansible-documentation` dir and `ansible` dir at the same level but different directories.  So a git checkout would be:
```
samccann:ansible$ git checkout devel
samccann:ansible-documentation$ git checkout devel
```
Does this mean I need the `--docs` and` --core `options?
`

x1101 replied:
No, having `ansible/ansible` and `ansible/ansible-documentation` in the same directory is what the tool expects, so you wouldn't need to use the `--docs` and `--core` options. if you had them in different locations for some reason (say, you had docs in one place, but did a checkout of core in a temp location) then you'd need to specify one or the other specific paths. 

I'd love some advice on how to make that easier to understand from the documentation.

samccann commented on README.md:
```diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Lastly, run the appropriate command(s) with the tagger script.
+
+Search for tags in the `ansible-core` repository that are missing from the `ansible-documentation` repository.
```
```suggestion
From  your `ansible-documentation directory`, search for tags in the `ansible-core` repository that are missing from the `ansible-documentation` repository.
```

samccann commented on README.md:
```diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+When a tag is created in the [`ansible/ansible`](https://github.com/ansible/ansible) repository for a release or release candidate, a corresponding tag should be created in this `ansible-documentation` repository.
+
+First, install the additional tagging dependencies as follows, creating or activating a `venv` as needed:
```
```suggestion
First, install the additional tagging dependencies from this repository as follows, creating or activating a `venv` as needed:
```

x1101 committed:
Update README.md

Co-authored-by: Sandra McCann <samccann@redhat.com>

x1101 committed:
Update README.md

Co-authored-by: Sandra McCann <samccann@redhat.com>

x1101 committed:
Cleanup of checkout directory description for clarity

samccann reviewed (approved):
LGTM. I think if I had to do it, I could follow these instructions. Thanks!

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+First, install the additional tagging dependencies from this repository as follows, creating or activating a `venv` as needed:
+
+```bash
````
I used the extra space is used in the rest of the README

```suggestion
``` bash
```

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+pip install -r hacking/tagger/requirements.txt
+```
+
+Next, ensure that you have both [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out. The tool assumes that both checkouts have the same parent directory. You can set different paths to your checkouts with the `--docs` and `--core` options if you have them setup another way.
````
I changed `both` -> `the` and `setup` (noun) -> `set up` (verb) and added a semantic line break.

```suggestion
Next, ensure that you have the [`ansible/ansible`](https://github.com/ansible/ansible) and [`ansible/ansible-documentation`](https://github.com/ansible/ansible-documentation) repositories checked out.
The tool assumes that both checkouts have the same parent directory. You can set different paths to your checkouts with the `--docs` and `--core` options if you have them set up another way.
```

gotmax23 commented:
I'll give this a quick review. Thanks

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+Lastly, run the appropriate command(s) with the tagger script.
+
+From  your `ansible-documentation` directory, search for tags in the `ansible-core` repository that are missing from the `ansible-documentation` repository.
+
+```bash
+./hacking/tagger/tag.py new-tags
+```
````
Running the `tag` subcommand directly should do the job. It just exits normally if there is nothing to tag. `new-tags` is mainly there for debugging and informational purposes, so I'm not sure it makes sense to demonstrate it here. The goal of the script is to provide a *single* idempotent command that does everything for all the actively supported branches without the user having to perform extra steps or decision making.

x1101 replied:
Thanks for the insight on this. I'll tune this up to align with that.

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+
+If the preceding command shows that there are new `ansible-core` tags, create corresponding ones for the `ansible-documentation` repository as follows:
+
+```bash
````
```suggestion
``` bash
```

gotmax23 commented on README.md:
````diff
@@ -136,3 +136,40 @@ If you do not have Python 3.10 installed, you can use root-less podman with a Py
+If the preceding command shows that there are new `ansible-core` tags, create corresponding ones for the `ansible-documentation` repository as follows:
+
+```bash
+# The tagger scripts assumes "origin" as the upstream remote.
+./hacking/tagger/tag.py tag
+
+# If you use a different upstream remote, specify the name.
+./hacking/tagger/tag.py --remote <name> tag 
+
+# If your core repo is not in the same filesystem location, specify the path.
+./hacking/tagger/tag.py --core <path> tag
````
I like the different examples with comments. Thank you.

x1101 committed:
Update README.md

Co-authored-by: Maxwell G <maxwell@gtmx.me>

x1101 committed:
Update README.md

Co-authored-by: Maxwell G <maxwell@gtmx.me>

x1101 committed:
Update README.md

Co-authored-by: Maxwell G <maxwell@gtmx.me>

x1101 committed:
Updates for clarity around use of the tagger script (fewer steps needed)

samccann commented:
Thanks @x1101 for the Ansible docs fix!
